.clang-format

```diff
@@ -0,0 +1,173 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```

Comment:
What's stopping us from using the Google defaults to make this file a lot more concise?

Reply:
I've changed the workflow to generate the Google style format file dynamically (thus removing the need for this file at all). Do you think that's fine?